common/djangoapps/student/views.py

```diff
@@ -2054,10 +2054,7 @@ def create_account_with_params(request, params):
         not settings.FEATURES.get('SKIP_EMAIL_VALIDATION', None) and
         not settings.FEATURES.get('AUTOMATIC_AUTH_FOR_TESTING') and
         not (do_external_auth and settings.FEATURES.get('BYPASS_ACTIVATION_EMAIL_FOR_EXTAUTH')) and
-        not (
-            third_party_provider and third_party_provider.skip_email_verification and
-            user.email == running_pipeline['kwargs'].get('details', {}).get('email')
-        )
+        not (third_party_provider and third_party_provider.skip_email_verification)
```

Comment:
I think we want to leave the email check in there if this is not an IdP that makes use of the OData API. Perhaps we could check the type of IdP and ignore the email check if we are dealing with a SF IdP. Something like the following?
```
not (
    third_party_provider and third_party_provider.skip_email_verification and
    (
        user.email == running_pipeline['kwargs'].get('details', {}).get('email') or
        third_party_provider.identity_provider_type == SAP_SUCCESSFACTORS_SAML_KEY
    )
)
```

Reply:
Done. I have also refactored code a bit to make it less confusing and added tests for the changes.